### DISCUSSION TITLE
Miller 6 and jsonlines: cut field by name

### DISCUSSION THREAD

**Discussion by aborruso** [johnkerl/miller#1013] [Q&A] [answered]
Hi,
I have a very stupid question

I have this input

```
{"name":"andrea","info":{"age":49,"city":"Palermo"}}
```

If I cat it I have `name=andrea,info.age=49,info.city=Palermo`.

If I want to cut `info.age` field, running `mlr --ijsonl  cut -f 'info.age' i.jsonl` I have empty result.
What's the field name I must use?

Thank you

**Comment by johnkerl** [comment, marked as answer]
Hi @aborruso ! This is more of #310 #763 #948.

There isn't a field called `info.age` until the final (implicit) flatten (https://miller.readthedocs.io/en/latest/flatten-unflatten/). There's a field called `info` which is a map which contains a key `age`. The final (implicit) flatten is to prepare for CSV output.

You can work around this by doing the flatten explicitly:

```
$ mlr --ijsonl flatten then cut -f 'info.age' i.jsonl
info.age=49
```

I do intend (#948) to let non-DSL verbs like `cut` be able to treat things like `info.age` syntactically -- this needs to be coded up.
